storage/csv/datasource.go

```diff
@@ -119,12 +119,16 @@ type RecordStream struct {
 	batchSize       int
 }
 
-func (rs *RecordStream) Close() error {
-	err := rs.file.Close()
-	if err != nil {
+func (rs *RecordStream) Close(ctx context.Context) error {
```

Comment:
I think this won't shut off the worker?

Reply:
I think it would be better if there was a boolean variable $running$ in badger. It would be set to false. Then, when the worker noticed it, it would shut itself down and delete the underlying storage.

Reply:
Ah no, Close is not transactional because of DropAll.

Reply:
So we can just cancel some context.

Reply:
Like with the subscriptions.